### ISSUE TITLE
Critical dependency for both create-next-app and create-react-app latest version

### ISSUE THREAD

**Issue by albusphan** [vladmandic/face-api#87]
**Issue Description**
Screenshot on nextjs
<img width="798" alt="Screen Shot 2021-12-01 at 20 38 19" src="https://user-images.githubusercontent.com/58453812/144244295-a0852b19-45bb-4dbd-a405-9c9d4c1911f5.png">

**Steps to Reproduce**
1. create brand new project with create-next-app or create-react-app
2. install @vladmandic/face-api
3. import to project.
3. start the project, the warning message appears.

**Expected Behavior**
No warn in start/build
**Environment

- Module version? 1.5.8
- Built-in demo or custom code?
- Type of module used (e.g. `js`, `esm`, `esm-nobundle`)? js
- Browser or NodeJS and version (e.g. NodeJS 14.15 or Chrome 89)?
- OS and Hardware platform (e.g. Windows 10, Ubuntu Linux on x64, Android 10)? macos
- Packager (if any) (e.g, webpack, rollup, parcel, esbuild, etc.)? webpack

**Additional**

- For installation or startup issues include your `package.json`
- For usage issues, it is recommended to post your code as [gist](https://gist.github.com/)

**Comment by vladmandic** [comment]
is this happening in latest version only?  
most likely due to your project doing a strict typedef checks  
try installing `npm install @tensorflow/tfjs` so its type definitions are available during your **nextjs** build process

**Comment by albusphan** [comment]
hi @vladmandic , yes it happens on previous version (1.5.7)
Both create-react-app an nextjs

**Comment by vladmandic** [comment]
warning message comes from bundle step (using `webpack`) which doesnt like that `face-api` is universal for both browser and nodejs environments. its perfectly safe and warning is fully ignorable.
i'll see if i can somehow get resolve it so warning doesnt appear.

**Comment by albusphan** [comment]
Thank you @vladmandic , one more question
When using face-api, the build time took about 2mins, but when I switch to your lib, it took more than 15mins
Is this intended?

**Comment by vladmandic** [comment]
`tfjs` 3.x is much more complex than `tfjs` 1.x used in original, but it should never result in such a huge difference  
i'd even say any build that takes more than 2min there is something wrong  

i just run a quick test with `webpack` and it took less than a minute  

can you share your app?

**Comment by albusphan** [comment]
hi @vladmandic 
quite ez to test the build time
1. npx create-react-app
2. install your lib
3. import your lib to App.js
4. build and wait for more than 800s 😄 

5. Rebuild and it took 18s (magic)

FYI: I'm using github actions so it always take 800s to build new version 😄

**Comment by vladmandic** [comment]
ok, i'll take a look, estimate early next week as i'm out for the rest of this week

**Comment by albusphan** [comment]
Thank you so much @vladmandic

**Comment by vladmandic** [comment]
# Issue: Slow React production build

`npm run start` runs dev build and is fast, ready in few seconds
`npm run build` runs production build and is really slow, ready in few minutes (rebuild is fast as it relies on cached results)

Issue is that React production build runs a `webpack` plugin `ForkTsCheckerWebpackPlugin` which has major issues so what ends up is `fork-ts-checker-webpack-plugin/lib/service.js` spawned process consuming tons of CPU until it times out  
And development build does not run `ForkTsCheckerWebpackPlugin` at all  

Anyhow, issue with `ForkTsCheckerWebpackPlugin` happens when React project size is over specific threshold (just happens that old FaceAPI is below threshold and new one is not). There are tons of reports on the net that same happens even when importing simple CSS icon sets, so it has nothing to do with actual checking  

I consider this a **major bug** in `react-scripts`  
And I'm not the only one, check out <https://github.com/facebook/create-react-app/issues?q=ForkTsCheckerWebpackPlugin>

Unfortunately, due to how React handles its internal architecture, there is no easy way to disable `ForkTsCheckerWebpackPlugin` - that is an open request for React for quite some time and FB is ignoring it  

(in React, there is no even a way to perform a user-level configuration of **webpack** without doing full `react-scripts eject` as messing with `node_modules/react-scripts/config/webpack.config.js` is definitely not recommended)

Whats even worse, event latest React is using an obsolete version of `fork-ts-checker-webpack-plugin` **4.1.6** which is over 2 years old! While latest published one is **6.5.0**

Sorry, not much here I can do...

FYI, I did a plugin dump during `dev` and `prod` runs of webpack, this is the configuration that Rect is using:

```json
{
  config: {
    mode: 'production',
    plugins: [
      [HtmlWebpackPlugin],
      [InlineChunkHtmlPlugin],
      [InterpolateHtmlPlugin],
      [ModuleNotFoundPlugin],
      [DefinePlugin],
      [MiniCssExtractPlugin],
      [ManifestPlugin],
      [IgnorePlugin],
      [ForkTsCheckerWebpackPlugin],
      [ESLintWebpackPlugin]
    ],
}
```

```json
{
  config: {
    mode: 'development',
    plugins: [
      [HtmlWebpackPlugin],
      [InterpolateHtmlPlugin],
      [ModuleNotFoundPlugin],
      [DefinePlugin],
      [HotModuleReplacementPlugin],
      [ReactRefreshPlugin],
      [CaseSensitivePathsPlugin],
      [WatchMissingNodeModulesPlugin],
      [ManifestPlugin],
      [IgnorePlugin],
      [ESLintWebpackPlugin]
    ],
}
```

**Comment by vladmandic** [comment]
btw, to close the loop on `critical dependency` issue:

it happens because of code in `src/env/createFileSystem.ts`:
```js
  if (!fs && isNodejs()) {
    try {
      fs = require('fs');
    } catch (err) {
      requireFsError = (err as any).toString();
    }
  }
```

which is clearly not executed in browser environments, but `webpack` anyhow tries to resolve `fs` which doesn't exist,  
so it throws irrelevant warning:
> Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
